spec/globignore.test.sh

```diff
@@ -0,0 +1,159 @@
+
+## oils_failures_allowed: 14
+## compare_shells: bash
+
+#### Don't glob flags on file system with GLOBIGNORE
+# This is a bash-specific extension.
+expr $0 : '.*/osh$' >/dev/null && exit 99  # disabled until cd implemented
+touch _tmp/-n _tmp/zzzzz
+cd _tmp  # this fail in osh
+GLOBIGNORE=-*:zzzzz  # colon-separated pattern list
+echo -* hello zzzz?
+## stdout-json: "-* hello zzzz?\n"
+## N-I dash/mksh/ash stdout-json: "hello zzzzz"
+## status: 0
+
+#### basic star case -> ignore files with txt extension
+touch {basic.md,basic.txt}
+GLOBIGNORE=*.txt
+echo *.*
+## STDOUT:
+basic.md
+## END
+
+#### basic question mark case -> ignore txt files with one char filename
+touch {1.txt,10.txt}
+GLOBIGNORE=?.txt
```

Comment:
I think the `?.txt` and `*.txt` will not be separate in the code -- it all boils down to a glob() call?  But it's OK to have some redundant tests -- it can serve as documentation and help us understand the feature